extension/README.md.args.mjs

```diff
@@ -0,0 +1,753 @@
+
+</details>
+
+<details markdown='1'><summary>Question Seven</summary>
+
+> The DEX also lost liquidity, how should we update \`totalLiquidity\`?
+
+</details>
+
+<details markdown='1'><summary>Question Eight</summary>
+
+> How do pay the user the value of \`ethWithdrawn\`?
+
+</details>
+
+<details markdown='1'><summary>Question Nine</summary>
+
+> How do we give them their tokens?
+
+</details>
+
+<details markdown='1'><summary>Question Ten</summary>
+
+> We have an event to emit, which one?
+
+</details>
+
+<details markdown='1'><summary>Question Eleven</summary>
+
+> Last, what are we returning?
+
+</details>
+
+<details markdown='1'><summary>👩🏽‍🏫 Solution Code </summary>
+
+\`\`\`
+
+function withdraw(uint256 amount) public returns(uint256 ethAmount, uint256 tokenAmount) {
+  require(liquidity[msg.sender] >= amount, "withdraw: sender does not have enough liquidity to withdraw.");
+        uint256 ethReserve = address(this).balance;
+        uint256 tokenReserve = token.balanceOf(address(this));
+        uint256 ethWithdrawn;
+
+  ethWithdrawn = amount * ethReserve / totalLiquidity;
+
+        uint256 tokenAmount = amount * tokenReserve / totalLiquidity;
+  liquidity[msg.sender] -= amount;
+  totalLiquidity -= amount;
+  (bool sent, ) = payable(msg.sender).call{ value: ethWithdrawn } ("");
+  require(sent, "withdraw(): revert in transferring eth to you!");
+  require(token.transfer(msg.sender, tokenAmount));
+        emit LiquidityRemoved(msg.sender, amount, tokenAmount, ethWithdrawn);
+  return (ethWithdrawn, tokenAmount);
+}
+
+\`\`\`
+
+ </details>
+
+ </details>
+
+🚨 Take a second to understand what these functions are doing if you pasted them into your \`DEX.sol\` file in packages/hardhat/contracts:
+
+### 🥅 Goals / Checks
+
+- [ ] 💧 Deposit liquidity, and then check your liquidity amount through the mapping in the debug tab. Has it changed properly? Did the right amount of assets get deposited?
+
+- [ ] 🧐 What happens if you \`deposit()\` at the beginning of the deployed contract, then another user starts swapping out for most of the balloons, and then you try to withdraw your position as a liquidity provider? Answer: you should get the amount of liquidity proportional to the ratio of assets within the isolated liquidity pool. It will not be 1:1.
+
+---
+
+## Checkpoint 6: UI 🖼
+
+Cool beans! Your front-end should be showing something like this now!
+
+![reserve-graph-updating](https://github.com/scaffold-eth/se-2-challenges/assets/55535804/ef5ddc5f-2fc8-4bb6-910f-d6c17e579a74)
+
+Now, a user can just enter the amount of ETH or tokens they want to swap and the chart will display how the price is calculated. The user can also visualize how larger swaps result in more slippage and less output asset.
+
+### ⚔️ Side Quests
+
+- [ ] In \`packages\nextjs\app\events\page.tsx\` implement an event and emit for the \`approve()\` function to make it clear when it has been executed.
```

Comment:
\`packages\nextjs\app\events\page.tsx\`  this is getting rendered badly in my instance readme:

```
In `packages
extjsappeventspage.tsx`
```

Reply:
I think there was a backslash escaping problem here, was still seeing it wrong. 

As a quick fix just changed the backslash for regular slash in https://github.com/scaffold-eth/se-2-challenges/pull/245/commits/a9a4a12be7e17e76accf01c9388e3b649799e77b since we're using regular slashes in other folders.